lib/hiera/backend/eyaml/plugins.rb

```diff
@@ -31,7 +31,8 @@ def self.find
           [index].flatten.each do |source|
             specs = Gem::VERSION >= "1.6.0" ? source.latest_specs(true) : source.latest_specs
 
-            specs.each do |spec|
+            specs.each do |stub|
+              spec = Gem::VERSION >= "3.2.33" ? stub.to_spec : stub
```

Comment:
```
irb(main):002:0> Gem::VERSION
=> "3.0.8"
irb(main):003:0> Gem::VERSION.class
=> String
```

Comparing string is probably not safe:

```
irb(main):004:0> "3.10.0" < "3.2.33"
=> true
irb(main):005:0> Gem::Version.new("3.10.0") < Gem::Version.new("3.2.33")
=> false
```

```suggestion
              spec = Gem::Version.new(Gem::VERSION) >= Gem::Version.new("3.2.33") ? stub.to_spec : stub
```

Reply:
Done, thanks! I've also fixed the other version comparisons in this file too.